apt-hook/hook.cc

```diff
@@ -123,9 +125,19 @@ static void check_esm_upgrade(pkgCache::PkgIterator pkg, pkgPolicy *policy, resu
 	 if (pf.File().Archive() != 0 && pf.File().Origin() == std::string("UbuntuESM"))
 	 {
 	    if (policy->GetPriority(pf.File()) == -32768)
-	       res.disabled_esms++;
+	       res.disabled_esms_i++;
 	    else
-	       res.enabled_esms++;
+	       res.enabled_esms_i++;
+
+	    return;
+	 }
+
+	 if (pf.File().Archive() != 0 && pf.File().Origin() == std::string("UbuntuESMApps"))
+	 {
+	    if (policy->GetPriority(pf.File()) == -32768)
```

Comment:
@murraybd Is this disabled check and counter needed?

 I was under the impression that only "ESM Infra" apt sources (on trusty only) would be installed without proper authentication which results in the -32768 value. In my mind, no other Ubuntu release would need this logic, and specifically not UbuntuESMApps.   The only time we would use this is if UAClient decides to install apt sources without credentials for ESM Apps to get 'visibility' to those potential packages.


That said, it probably doesn't hurt much to have this check and count in here, that way we can support advertising ESM Apps on disabled machines if we need to.

Reply:
Here's the SPEC reference that says we don't pre-advertise UA Apps or UA infra packages on Xenial+ https://docs.google.com/document/d/19_bTeagOuY9Wd727vtLskP1rkY5FwzaFXi3b_R5qa7g/edit#